### PR TITLE
Fix : Avoid re-rendering smartsheet

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 1]
```diff
@@ -413,7 +413,9 @@ const expandedFormOnRowIdDlg = computed({
 provide(ReloadRowDataHookInj, reloadViewDataHook)
 
 // trigger initial data load in grid
-reloadViewDataHook.trigger()
+// reloadViewDataHook.trigger()
+
+watch(meta, () => reloadViewDataHook.trigger(), { immediate: true })
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/Grid.vue:418]: With this, wouldn't we trigger the loading twice in a row when the view changes?

Further up in `Grid.vue` there's another watcher:

```ts
watch(
  () => view.value?.id,
  async (next, old) => {
    if (next && old && next !== old) {
      await loadData()
    }
  },
  { immediate: true },
)
```

File: packages/nc-gui/components/smartsheet/sidebar/MenuTop.vue [hunk 2]
```diff
@@ -214,6 +214,12 @@ function openDeleteDialog(view: Record<string, any>) {
     close(1000)
   }
 }
+
+watch(views, (nextViews) => {
+  if(nextViews?.length && (!activeView.value || !nextViews.includes(activeView.value))){
+    activeView.value = nextViews[0]
+  }
+})
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/sidebar/MenuTop.vue:222]: Same thing is triggered in `sidebar/index.vue` watcher, isn't it?

```ts
watch(
  [views, () => route.params.viewTitle],
  () => {
    // watcher	

    /** if active view is not found, set it to first view */
    if (!activeView.value && nextViews.length) {
      activeView.value = nextViews[0]
    }
})
```

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 3]
```diff
@@ -412,8 +411,7 @@ reloadViewDataHook.trigger()
               </th>
             </tr>
           </thead>
-          <!-- this prevent select text from field if not in edit mode -->
-          <tbody @selectstart.prevent>
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/Grid.vue:416]: should this be removed? I think it was something that was added for multi-select?

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 4]
```diff
@@ -462,17 +460,13 @@ reloadViewDataHook.trigger()
                     :key="columnObj.id"
                     class="cell relative cursor-pointer nc-grid-cell"
                     :class="{
-                      active:
-                        (isUIAllowed('xcDatatableEditable') && selected.col === colIndex && selected.row === rowIndex) ||
-                        (isUIAllowed('xcDatatableEditable') && selectedRange(rowIndex, colIndex)),
+                      active: isUIAllowed('xcDatatableEditable') && selected.col === colIndex && selected.row === rowIndex,
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/Grid.vue:463]: I think we can replace all the `isUIAllowed('xcDatatableEditable')` calls with the computed property on top of this file called `hasEditPermission`

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 5]
```diff
@@ -462,17 +460,13 @@ reloadViewDataHook.trigger()
...
                     }"
                     :data-key="rowIndex + columnObj.id"
                     :data-col="columnObj.id"
                     :data-title="columnObj.title"
                     @click="selectCell(rowIndex, colIndex)"
                     @dblclick="makeEditable(row, columnObj)"
-                    @mousedown="startSelectRange($event, rowIndex, colIndex)"
-                    @mouseover="selectBlock(rowIndex, colIndex)"
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/Grid.vue:475]: Why was this removed? It's for multi-select

File: packages/nc-gui/components/smartsheet/Gallery.vue [hunk 6]
```diff
@@ -159,6 +159,12 @@ onMounted(async () => {
 
 // provide view data reload hook as fallback to row data reload
 provide(ReloadRowDataHookInj, reloadViewDataHook)
+
+watch(view, (nextView) => {
+  if (nextView?.type === ViewTypes.FORM) {
+    reloadEventHook.trigger()
+  }
+})
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/Gallery.vue:167]: if the view type is form, then the Gallery component will not be mounted anyway, so this seems redundant.

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

File: packages/nc-gui/components/smartsheet/Grid.vue (hunk 3 above)
Comment (pranavxc) [reply packages/nc-gui/components/smartsheet/Grid.vue:416]: Something wrong happened while rebasing, will revert these changes :+1:

File: packages/nc-gui/pages/[projectType]/[projectId]/index.vue [hunk 7]
```diff
@@ -491,10 +491,10 @@ onBeforeUnmount(reset)
       </a-layout-sider>
     </template>
 
-    <div :key="$route.fullPath.split('?')[0]">
+    <div>
       <LazyDashboardSettingsModal v-model="dialogOpen" :open-key="openDialogKey" />
 
-      <NuxtPage />
+      <NuxtPage :transition="false" :page-key="$route.params.projectId" />
```
Comment (bcakmakoglu) [inline packages/nc-gui/pages/[projectType]/[projectId]/index.vue:497]: why transition false?

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

File: packages/nc-gui/pages/[projectType]/[projectId]/index.vue (hunk 7 above)
Comment (pranavxc) [reply packages/nc-gui/pages/[projectType]/[projectId]/index.vue:497]: It's leading to some issues if we close and open tabs too quickly, which is only reproducible in cypress. In cypress we already added some delay to take care of the issue, so we can revert if needed.

Comment (bcakmakoglu) [reply packages/nc-gui/pages/[projectType]/[projectId]/index.vue:497]: Ok 👍

**Review by bcakmakoglu — APPROVED** [review]
